src/main/java/us/kbase/auth/client/AuthClient.java

```diff
@@ -179,4 +186,53 @@ public AuthToken validateToken(final String token) throws IOException, AuthExcep
 		return authToken;
 	}
 
+	private void checkToken(final String token) {
+		if (token == null || token.trim().isEmpty()) {
+			throw new IllegalArgumentException("token must be a non-whitespace string");
+		}
+	}
+	
+	/** Check if usernames are valid accounts in the auth service.
+	 * @param users the list of usernames to check. If they contain any invalid characters
+	 * (e.g. anything other than a-z, 0-9, or _, an exception will be thrown.
+	 * @param token any valid auth token.
+	 * @return a mapping of each username to whether it's valid or not.
+	 * @throws IOException if an IOException occurs communicating with the auth service.
+	 * @throws AuthException if an auth exception occurs communicating with the auth service.
+	 */
+	public Map<String, Boolean> isValidUserName(final List<String> users, final String token)
+			throws IOException, AuthException {
+		// theoretically someone could submit hundreds of users and hit the url size limit
+		// don't worry about that for now.
+		checkToken(token);
+		if (users == null || users.isEmpty()) {
+			throw new IllegalArgumentException("users cannot be null or empty");
+		}
+		final List<String> badlist = new LinkedList<String>();
+		final Map<String, Boolean> result = new HashMap<>();
+		for (String user: users) {
+			if (user == null || user.trim().isEmpty()) {
+				throw new IllegalArgumentException("each user must be a non-whitespace string");
+			}
+			user = user.trim();
+			final Matcher m = INVALID_USERNAME.matcher(user);
+			if (m.find()) {
+				// this matches the prior behavior, but later could have an validity enum
+				// one for bad chars, one for does not exist
+				throw new IllegalArgumentException(
+						"username " + user + " has invalid character: " + m.group(0));
+			}
+			if (userCache.hasString(user)) {
+				result.put(user, true);
+			} else {
+				badlist.add(user);
+			}
+		}
+		final URI target = rootURI.resolve("api/V2/users/?list=" + String.join(",", badlist));
```

Comment:
Let's see if my understanding is correct. 

`badlist` consists gibberish usernames and expired ones. And you call `api/V2/users/` to distinguish them?

Reply:
badlist contains any usernames that don't exist in the cache and are valid usernames, and the endpoint returns extant usernames mapped to their display names

Reply:
Saw your text examples. Make sense now. Thanks!